cuda/base/cusparse_bindings.hpp

```diff
@@ -504,101 +623,119 @@ GKO_BIND_CUSPARSE64_CSR2HYB(ValueType, detail::not_implemented);
 #undef GKO_BIND_CUSPARSE64_CSR2HYB
 
 
+#endif
+
+
+#if defined(CUDA_VERSION) && (CUDA_VERSION < 11000)
+
+template <typename ValueType, typename IndexType>
+inline void transpose(cusparseHandle_t handle, size_type m, size_type n,
+                      size_type nnz, const ValueType *OrigValA,
+                      const IndexType *OrigRowPtrA,
+                      const IndexType *OrigColIndA, ValueType *TransValA,
+                      IndexType *TransRowPtrA, IndexType *TransColIndA,
+                      cusparseAction_t copyValues,
+                      cusparseIndexBase_t idxBase) GKO_NOT_IMPLEMENTED;
+
+// Cusparse csr2csc use the order (row_inx, col_ptr) for csc, so we need to
+// switch row_ptr and col_idx of transposed csr here
 #define GKO_BIND_CUSPARSE_TRANSPOSE32(ValueType, CusparseName)                \
-    inline void transpose(cusparseHandle_t handle, size_type m, size_type n,  \
-                          size_type nnz, const ValueType *OrigValA,           \
-                          const int32 *OrigRowPtrA, const int32 *OrigColIndA, \
-                          ValueType *TransValA, int32 *TransRowPtrA,          \
-                          int32 *TransColIndA, cusparseAction_t copyValues,   \
-                          cusparseIndexBase_t idxBase)                        \
+    template <>                                                               \
+    inline void transpose<ValueType, int32>(                                  \
+        cusparseHandle_t handle, size_type m, size_type n, size_type nnz,     \
+        const ValueType *OrigValA, const int32 *OrigRowPtrA,                  \
+        const int32 *OrigColIndA, ValueType *TransValA, int32 *TransRowPtrA,  \
+        int32 *TransColIndA, cusparseAction_t copyValues,                     \
+        cusparseIndexBase_t idxBase)                                          \
     {                                                                         \
         GKO_ASSERT_NO_CUSPARSE_ERRORS(                                        \
             CusparseName(handle, m, n, nnz, as_culibs_type(OrigValA),         \
                          OrigRowPtrA, OrigColIndA, as_culibs_type(TransValA), \
-                         TransRowPtrA, TransColIndA, copyValues, idxBase));   \
+                         TransColIndA, TransRowPtrA, copyValues, idxBase));   \
```

Comment:
Why did this change ?

Reply:
cusparseCsr2csc uses csc by (CscVal, CscRowInd, CscColPtr) - (1)
cusparseCsr2cscEx2 uses csc by (CscVal, CscColPtr, CscRowInd) - (2)
We use it mainly for transposing, so I think using transposed_csr by (transVal, transRowPtr, transColInd), i.e. (2),should be much familiar and not confusing.

Reply:
also another point is original we use transRowPtr as ColInd, which is wrong.